task/thermal/src/bsp/gimlet_b.rs

```diff
@@ -74,19 +83,43 @@ impl Bsp {
     }
 
     pub fn power_mode(&self) -> u32 {
-        match self.seq.get_state() {
-            Ok(p) => match p {
-                PowerState::A0PlusHP | PowerState::A0 | PowerState::A1 => {
-                    POWER_STATE_A0
+        let state = match self.seq.get_state() {
+            Ok(p) => p,
+            // If `get_state` failed, then enable all sensors except the M.2s
+            // (which can permanently lock up the system if they're read while
+            // unpowered). One of the sensors in the A0 / A2 domains will
+            // presumably fail and will drop us into failsafe eventually.
+            Err(_) => return POWER_STATE_A0 | POWER_STATE_A2,
+        };
+        match state {
+            PowerState::A0PlusHP | PowerState::A0 | PowerState::A1 => {
+                // The M.2 devices are enabled separately from A0, so we check
+                // for them by asking their power controller. There's a
+                // potential TOCTOU race here, but we don't expect to power
+                // these down after the server comes up.
+                let dev = devices::max5970_m2(self.i2c_task);
+                let m = drv_i2c_devices::max5970::Max5970::new(&dev);
+                let mut out = POWER_STATE_A0;
+                match m.read_reg(drv_i2c_devices::max5970::Register::status3) {
+                    Ok(s) => {
+                        // pg[0]
+                        if s & 1 != 0 {
+                            out |= POWER_M2A;
+                        }
+                        // pg[1]
+                        if s & 2 != 0 {
+                            out |= POWER_M2B;
+                        }
+                    }
+                    // TODO: error handling here?
```

Comment:
Do we need any error handling? Locally it looks like if we fail to read this register for any reason we have to assume we can't access the M.2 devices, unless there are specific error codes we might be able to do something with (retry?)?